gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1308 @@
+            num_models should be a multiple of ensemble_workers.
+
+            Setting it to 0 or 1 will both use the nonmultiprocessing version. Default: 1
+        memory_friendly_ttda : boolean, optional
+            If True, the models in the ensemble are deleted after training and only a concatenation of each model's
+            topic term distribution (called ttda) is kept to save memory.
+
+            Defaults to True. When False, trained models are stored in a list in self.tms, and no models that are not
+            of a gensim model type can be added to this ensemble using the add_model function.
+
+            If False, any topic term matrix can be suplied to add_model.
+        min_samples : int, optional
+            Required int of nearby topics for a topic to be considered as 'core' in the CBDBSCAN clustering.
+        masking_method : str, optional
+            Choose one of "mass" (default) or "rank" (percentile, faster).
+
+            For clustering, distances between topic-term distributions are asymmetric.  In particular, the distance
+            (technically a divergence) from distribution A to B is more of a measure of if A is contained in B.  At a
+            high level, this involves using distribution A to mask distribution B and then calculating the cosine
+            distance between the two.  The masking can be done in two ways:
+
+            1. mass: forms mask by taking the top ranked terms until their cumulative mass reaches the
+            'masking_threshold'
+
+            2. rank: forms mask by taking the top ranked terms (by mass) until the 'masking_threshold' is reached.
+            For example, a ranking threshold of 0.11 means the top 0.11 terms by weight are used to form a mask.
+
+        masking_threshold : float, optional
+            Default: None, which uses ``0.95`` for "mass", and ``0.11`` for masking_method "rank".  In general, too
+            small a mask threshold leads to inaccurate calculations (no signal) and too big a mask leads to noisy
+            distance calculations.  Defaults are often a good sweet spot for this hyperparameter.
+        distance_workers : int, optional
+            When ``distance_workers`` is ``None``, it defaults to ``os.cpu_count()`` for maximum performance. Default is
+            1, which is not multiprocessed. Set to ``> 1`` to enable multiprocessing.
+        **gensim_kw_args
+            Parameters for each gensim model (e.g. :py:class:`gensim.models.LdaModel`) in the ensemble.
+
+        """
+        # INTERNAL PARAMETERS
+        # Set random state
+        # nps max random state of 2**32 - 1 is too large for windows:
+        self._MAX_RANDOM_STATE = np.iinfo(np.int32).max
+
+        # _COSINE_DISTANCE_CALCULATION_THRESHOLD is used so that cosine distance calculations can be sped up by skipping
+        # distance calculations for highly masked topic-term distributions
+        self._COSINE_DISTANCE_CALCULATION_THRESHOLD = 0.05
```

Comment:
Is there a point to making this a class member as opposed to a module-level "constant"?